source/FreeRTOS_TCP_IP.c

```diff
@@ -594,7 +601,7 @@
             FreeRTOS_debug_printf( ( "Connect[%xip:%u]: next timeout %u: %u ms\n",
                                      ( unsigned ) pxSocket->u.xTCP.xRemoteIP.ulIP_IPv4, pxSocket->u.xTCP.usRemotePort,
                                      pxSocket->u.xTCP.ucRepCount, ( unsigned ) ulDelayMs ) );
-            pxSocket->u.xTCP.usTimeout = ( uint16_t ) ipMS_TO_MIN_TICKS( ulDelayMs );
+            pxSocket->u.xTCP.usTimeout = ( uint16_t ) ulDelayMs;
```

Comment:
Wondering why the conversion of time in milliseconds to rtos ticks (`ipMS_TO_MIN_TICKS`) was removed here. `pxSocket->u.xTCP.usTimeout` is expected to have time (in ticks).

Reply:
Good catch. I think I should drop this change.

I only tried to remove the clamping of 0 to 1, since I handled that in the millisecond domain, but this clamping is in the tick domain, so it's not the same.

Reply:
Cutting the commit down to the essential change,
here is the new commit 7c0e0605dc
and the total diff [84710cce..10c134d8](https://github.com/FreeRTOS/FreeRTOS-Plus-TCP/compare/84710cce..10c134d8).

I think it was a bit confusing that the variable is a TickType_t while holding milliseconds.
I would consider moving the call to ipMS_TO_MIN_TICKS to each place it is assigned instead.
I don't know if optimizers are generally smart enough to delete the clamping operation in its current place, but I would expect it on the constant expressions.